main.tf

```diff
@@ -76,62 +81,134 @@ resource "google_compute_url_map" "default" {
+  connection_draining_timeout_sec = lookup(each.value, "connection_draining_timeout_sec", null)
+  enable_cdn                      = lookup(each.value, "enable_cdn", false)
+  security_policy                 = var.security_policy
+  health_checks                   = [google_compute_health_check.default[each.key].self_link]
+
   dynamic "backend" {
-    for_each = var.backends[count.index]
+    for_each = toset(each.value["groups"])
     content {
       balancing_mode               = lookup(backend.value, "balancing_mode")
       capacity_scaler              = lookup(backend.value, "capacity_scaler")
       description                  = lookup(backend.value, "description")
       group                        = lookup(backend.value, "group")
       max_connections              = lookup(backend.value, "max_connections")
       max_connections_per_instance = lookup(backend.value, "max_connections_per_instance")
+      max_connections_per_endpoint = lookup(backend.value, "max_connections_per_endpoint")
       max_rate                     = lookup(backend.value, "max_rate")
       max_rate_per_instance        = lookup(backend.value, "max_rate_per_instance")
+      max_rate_per_endpoint        = lookup(backend.value, "max_rate_per_endpoint")
       max_utilization              = lookup(backend.value, "max_utilization")
     }
   }
-  health_checks   = [
-    google_compute_http_health_check.default[count.index].self_link]
-  security_policy = var.security_policy
-  enable_cdn      = var.cdn
-}
 
-resource "google_compute_http_health_check" "default" {
-  project      = var.project
-  count        = length(var.backend_params)
-  name         = "${var.name}-backend-${count.index}"
-  request_path = split(",", var.backend_params[count.index])[0]
-  port         = split(",", var.backend_params[count.index])[2]
+  # TODO:
+  #  dynamic "cdn_policy" {
+  #   for_each = [lookup(each.value, "cdn_policy", {})]
+  #   content {
+  #     signed_url_cache_max_age_sec = cdn_policy.value.signed_url_cache_max_age_sec
+  #     dynamic "cache_key_policy" {
+  #       for_each =  [lookup(cdn_policy.value, "cache_key_policy", {})]
+  #       content {
+  #         include_host = true
+  #       }
+  #     }
+  #   }
+  # }
+
+  depends_on = [
+    google_compute_health_check.default
+  ]
 }
 
-resource "google_compute_firewall" "default-hc" {
-  count         = length(var.firewall_networks)
-  project       = length(var.firewall_networks) == 1 && var.firewall_projects[0] == "default" ? var.project : var.firewall_projects[count.index]
-  name          = "${var.name}-hc-${count.index}"
-  network       = var.firewall_networks[count.index]
-  source_ranges = [
-    "130.211.0.0/22",
-    "35.191.0.0/16",
-    "209.85.152.0/22",
-    "209.85.204.0/22"]
-  target_tags   = var.target_tags
-
-  dynamic "allow" {
-    for_each = distinct(var.backend_params)
+resource "google_compute_health_check" "default" {
```

Comment:
It seems you're very closely wrapping `google_compute_health_check`. Why not just let users pass in their own health check `self_links` into the backend object?